// googlevideo@4.0.4/dist/protos/generated/video_streaming/onesie_proxy_status.js
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.5
//   protoc               v3.21.12
// source: video_streaming/onesie_proxy_status.proto
/* eslint-disable */
export const protobufPackage = "video_streaming";
export const OnesieProxyStatus = {
    UNKNOWN: 0, 0: "UNKNOWN",
    OK: 1, 1: "OK",
    DECRYPTION_FAILED: 2, 2: "DECRYPTION_FAILED",
    PARSING_FAILED: 3, 3: "PARSING_FAILED",
    MISSING_X_FORWARDED_FOR: 4, 4: "MISSING_X_FORWARDED_FOR",
    INVALID_X_FORWARDED_FOR: 5, 5: "INVALID_X_FORWARDED_FOR",
    INVALID_CONTENT_TYPE: 6, 6: "INVALID_CONTENT_TYPE",
    BACKEND_ERROR: 7, 7: "BACKEND_ERROR",
    CLIENT_ERROR: 8, 8: "CLIENT_ERROR",
    MISSING_CRYPTER: 9, 9: "MISSING_CRYPTER",
    RESPONSE_JSON_SERIALIZATION_FAILED: 10, 10: "RESPONSE_JSON_SERIALIZATION_FAILED",
    DECOMPRESSION_FAILED: 11, 11: "DECOMPRESSION_FAILED",
    JSON_PARSING_FAILED: 12, 12: "JSON_PARSING_FAILED",
    UNKNOWN_COMPRESSION_TYPE: 13, 13: "UNKNOWN_COMPRESSION_TYPE",
    UNRECOGNIZED: -1, "-1": "UNRECOGNIZED"
};
