// googlevideo@4.0.4/dist/protos/generated/video_streaming/sabr_context_update.js
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.5
//   protoc               v3.21.12
// source: video_streaming/sabr_context_update.proto
/* eslint-disable */
import { BinaryReader, BinaryWriter } from "@bufbuild/protobuf/wire";
export const protobufPackage = "video_streaming";
export const SabrContextUpdate_SabrContextScope = {
    UNKNOWN: 0, 0: "UNKNOWN",
    PLAYBACK: 1, 1: "PLAYBACK",
    REQUEST: 2, 2: "REQUEST",
    WATCH_ENDPOINT: 3, 3: "WATCH_ENDPOINT",
    CONTENT_ADS: 4, 4: "CONTENT_ADS",
    UNRECOGNIZED: -1, "-1": "UNRECOGNIZED"
};
export const SabrContextUpdate_SabrContextWritePolicy = {
    UNSPECIFIED: 0, 0: "UNSPECIFIED",
    OVERWRITE: 1, 1: "OVERWRITE",
    KEEP_EXISTING: 2, 2: "KEEP_EXISTING",
    UNRECOGNIZED: -1, "-1": "UNRECOGNIZED"
};
function createBaseSabrContextUpdate() {
    return { type: 0, scope: 0, value: new Uint8Array(0), sendByDefault: false, writePolicy: 0 };
}
export const SabrContextUpdate = {
    encode(message, writer = new BinaryWriter()) {
        if (message.type !== undefined && message.type !== 0) {
            writer.uint32(8).int32(message.type);
        }
        if (message.scope !== undefined && message.scope !== 0) {
            writer.uint32(16).int32(message.scope);
        }
        if (message.value !== undefined && message.value.length !== 0) {
            writer.uint32(26).bytes(message.value);
        }
        if (message.sendByDefault !== undefined && message.sendByDefault !== false) {
            writer.uint32(32).bool(message.sendByDefault);
        }
        if (message.writePolicy !== undefined && message.writePolicy !== 0) {
            writer.uint32(40).int32(message.writePolicy);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof BinaryReader ? input : new BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSabrContextUpdate();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 8) {
                        break;
                    }
                    message.type = reader.int32();
                    continue;
                }
                case 2: {
                    if (tag !== 16) {
                        break;
                    }
                    message.scope = reader.int32();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.value = reader.bytes();
                    continue;
                }
                case 4: {
                    if (tag !== 32) {
                        break;
                    }
                    message.sendByDefault = reader.bool();
                    continue;
                }
                case 5: {
                    if (tag !== 40) {
                        break;
                    }
                    message.writePolicy = reader.int32();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
};
function createBaseSabrContextValue() {
    return { timing: undefined, signature: new Uint8Array(0), field5: 0 };
}
export const SabrContextValue = {
    encode(message, writer = new BinaryWriter()) {
        if (message.timing !== undefined) {
            SabrContextValue_TimingInfo.encode(message.timing, writer.uint32(10).fork()).join();
        }
        if (message.signature !== undefined && message.signature.length !== 0) {
            writer.uint32(18).bytes(message.signature);
        }
        if (message.field5 !== undefined && message.field5 !== 0) {
            writer.uint32(40).int32(message.field5);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof BinaryReader ? input : new BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSabrContextValue();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.timing = SabrContextValue_TimingInfo.decode(reader, reader.uint32());
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.signature = reader.bytes();
                    continue;
                }
                case 5: {
                    if (tag !== 40) {
                        break;
                    }
                    message.field5 = reader.int32();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
};
function createBaseSabrContextValue_ContentInfo() {
    return { contentId: "", contentType: 0 };
}
export const SabrContextValue_ContentInfo = {
    encode(message, writer = new BinaryWriter()) {
        if (message.contentId !== undefined && message.contentId !== "") {
            writer.uint32(10).string(message.contentId);
        }
        if (message.contentType !== undefined && message.contentType !== 0) {
            writer.uint32(16).int32(message.contentType);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof BinaryReader ? input : new BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSabrContextValue_ContentInfo();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.contentId = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 16) {
                        break;
                    }
                    message.contentType = reader.int32();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
};
function createBaseSabrContextValue_TimingInfo() {
    return { timestampMs: "0", durationMs: 0, content: undefined };
}
export const SabrContextValue_TimingInfo = {
    encode(message, writer = new BinaryWriter()) {
        if (message.timestampMs !== undefined && message.timestampMs !== "0") {
            writer.uint32(8).int64(message.timestampMs);
        }
        if (message.durationMs !== undefined && message.durationMs !== 0) {
            writer.uint32(16).int32(message.durationMs);
        }
        if (message.content !== undefined) {
            SabrContextValue_ContentInfo.encode(message.content, writer.uint32(26).fork()).join();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof BinaryReader ? input : new BinaryReader(input);
        const end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSabrContextValue_TimingInfo();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 8) {
                        break;
                    }
                    message.timestampMs = reader.int64().toString();
                    continue;
                }
                case 2: {
                    if (tag !== 16) {
                        break;
                    }
                    message.durationMs = reader.int32();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.content = SabrContextValue_ContentInfo.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
};
