// googlevideo@4.0.4/dist/protos/generated/video_streaming/ump_part_id.js
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.5
//   protoc               v3.21.12
// source: video_streaming/ump_part_id.proto
/* eslint-disable */
export const protobufPackage = "video_streaming";
export const UMPPartId = {
    UNKNOWN: 0, 0: "UNKNOWN",
    ONESIE_HEADER: 10, 10: "ONESIE_HEADER",
    ONESIE_DATA: 11, 11: "ONESIE_DATA",
    ONESIE_ENCRYPTED_MEDIA: 12, 12: "ONESIE_ENCRYPTED_MEDIA",
    /** MEDIA_HEADER - Header for a media segment; includes sequence and timing information. */
    MEDIA_HEADER: 20, 
    /** MEDIA_HEADER - Header for a media segment; includes sequence and timing information. */
    20: "MEDIA_HEADER",
    /** MEDIA - Chunk of media segment data. */
    MEDIA: 21, 
    /** MEDIA - Chunk of media segment data. */
    21: "MEDIA",
    /** MEDIA_END - Indicates end of media segment; finalizes segment processing. */
    MEDIA_END: 22, 
    /** MEDIA_END - Indicates end of media segment; finalizes segment processing. */
    22: "MEDIA_END",
    CONFIG: 30, 30: "CONFIG",
    LIVE_METADATA: 31, 31: "LIVE_METADATA",
    HOSTNAME_CHANGE_HINT_DEPRECATED: 32, 32: "HOSTNAME_CHANGE_HINT_DEPRECATED",
    LIVE_METADATA_PROMISE: 33, 33: "LIVE_METADATA_PROMISE",
    LIVE_METADATA_PROMISE_CANCELLATION: 34, 34: "LIVE_METADATA_PROMISE_CANCELLATION",
    /** NEXT_REQUEST_POLICY - Server's policy for the next request; includes backoff time and playback cookie. */
    NEXT_REQUEST_POLICY: 35, 
    /** NEXT_REQUEST_POLICY - Server's policy for the next request; includes backoff time and playback cookie. */
    35: "NEXT_REQUEST_POLICY",
    USTREAMER_VIDEO_AND_FORMAT_METADATA: 36, 36: "USTREAMER_VIDEO_AND_FORMAT_METADATA",
    FORMAT_SELECTION_CONFIG: 37, 37: "FORMAT_SELECTION_CONFIG",
    USTREAMER_SELECTED_MEDIA_STREAM: 38, 38: "USTREAMER_SELECTED_MEDIA_STREAM",
    /** FORMAT_INITIALIZATION_METADATA - Metadata for format initialization; contains total number of segments, duration, etc. */
    FORMAT_INITIALIZATION_METADATA: 42, 
    /** FORMAT_INITIALIZATION_METADATA - Metadata for format initialization; contains total number of segments, duration, etc. */
    42: "FORMAT_INITIALIZATION_METADATA",
    /** SABR_REDIRECT - Indicates a redirect to a different streaming URL. */
    SABR_REDIRECT: 43, 
    /** SABR_REDIRECT - Indicates a redirect to a different streaming URL. */
    43: "SABR_REDIRECT",
    /** SABR_ERROR - Indicates a SABR error; happens when the payload is invalid or the server cannot process the request. */
    SABR_ERROR: 44, 
    /** SABR_ERROR - Indicates a SABR error; happens when the payload is invalid or the server cannot process the request. */
    44: "SABR_ERROR",
    SABR_SEEK: 45, 45: "SABR_SEEK",
    /** RELOAD_PLAYER_RESPONSE - Directive to reload the player with new parameters. */
    RELOAD_PLAYER_RESPONSE: 46, 
    /** RELOAD_PLAYER_RESPONSE - Directive to reload the player with new parameters. */
    46: "RELOAD_PLAYER_RESPONSE",
    PLAYBACK_START_POLICY: 47, 47: "PLAYBACK_START_POLICY",
    ALLOWED_CACHED_FORMATS: 48, 48: "ALLOWED_CACHED_FORMATS",
    START_BW_SAMPLING_HINT: 49, 49: "START_BW_SAMPLING_HINT",
    PAUSE_BW_SAMPLING_HINT: 50, 50: "PAUSE_BW_SAMPLING_HINT",
    SELECTABLE_FORMATS: 51, 51: "SELECTABLE_FORMATS",
    REQUEST_IDENTIFIER: 52, 52: "REQUEST_IDENTIFIER",
    REQUEST_CANCELLATION_POLICY: 53, 53: "REQUEST_CANCELLATION_POLICY",
    ONESIE_PREFETCH_REJECTION: 54, 54: "ONESIE_PREFETCH_REJECTION",
    TIMELINE_CONTEXT: 55, 55: "TIMELINE_CONTEXT",
    REQUEST_PIPELINING: 56, 56: "REQUEST_PIPELINING",
    /** SABR_CONTEXT_UPDATE - Updates SABR context data; usually used for ads. */
    SABR_CONTEXT_UPDATE: 57, 
    /** SABR_CONTEXT_UPDATE - Updates SABR context data; usually used for ads. */
    57: "SABR_CONTEXT_UPDATE",
    /** STREAM_PROTECTION_STATUS - Status of stream protection; indicates whether attestation is required. */
    STREAM_PROTECTION_STATUS: 58, 
    /** STREAM_PROTECTION_STATUS - Status of stream protection; indicates whether attestation is required. */
    58: "STREAM_PROTECTION_STATUS",
    /** SABR_CONTEXT_SENDING_POLICY - Policy indicating which SABR contexts to send or discard in future requests. */
    SABR_CONTEXT_SENDING_POLICY: 59, 
    /** SABR_CONTEXT_SENDING_POLICY - Policy indicating which SABR contexts to send or discard in future requests. */
    59: "SABR_CONTEXT_SENDING_POLICY",
    LAWNMOWER_POLICY: 60, 60: "LAWNMOWER_POLICY",
    SABR_ACK: 61, 61: "SABR_ACK",
    END_OF_TRACK: 62, 62: "END_OF_TRACK",
    CACHE_LOAD_POLICY: 63, 63: "CACHE_LOAD_POLICY",
    LAWNMOWER_MESSAGING_POLICY: 64, 64: "LAWNMOWER_MESSAGING_POLICY",
    PREWARM_CONNECTION: 65, 65: "PREWARM_CONNECTION",
    PLAYBACK_DEBUG_INFO: 66, 66: "PLAYBACK_DEBUG_INFO",
    /** SNACKBAR_MESSAGE - Directive to show the user a notification message. */
    SNACKBAR_MESSAGE: 67, 
    /** SNACKBAR_MESSAGE - Directive to show the user a notification message. */
    67: "SNACKBAR_MESSAGE",
    UNRECOGNIZED: -1, "-1": "UNRECOGNIZED"
};
